### ISSUE TITLE
Converting units from Figma Tokens

### ISSUE THREAD

**Issue by djmtype** [amzn/style-dictionary#796]
I'm using the [Figma Tokens](https://www.figma.com/community/plugin/843461159747178978/Figma-Tokens) preset by example. 

All of these units for font sizes are unit-less but based on pixels. What's the best practice for converting these to rem values as I don't want to manually change the `fontSizes.body` value afterwards? 

I've also been having to use the [token-transformer](https://www.npmjs.com/package/token-transformer) plugin in order to computer these values before passing them to Style Dictionary.

```json
{
  "spacing": {
    "scale": {
      "value": "2",
      "type": "spacing"
    },
    "xs": {
      "value": "4",
      "type": "spacing"
    },
    "sm": {
      "value": "{spacing.xs} * {spacing.scale}",
      "type": "spacing"
    },
    "md": {
      "value": "{spacing.sm} * {spacing.scale}",
      "type": "spacing"
    },
    "lg": {
      "value": "{spacing.md} * {spacing.scale}",
      "type": "spacing"
    },
    "xl": {
      "value": "{spacing.lg} * {spacing.scale}",
      "type": "spacing"
    }
  },
  "sizing": {
    "scale": {
      "value": "1.5",
      "type": "sizing"
    },
    "xs": {
      "value": "4",
      "type": "sizing"
    },
    "sm": {
      "value": "{sizing.xs} * {sizing.scale}",
      "type": "sizing"
    },
    "md": {
      "value": "{sizing.sm} * {sizing.scale}",
      "type": "sizing"
    },
    "lg": {
      "value": "{sizing.md} * {sizing.scale}",
      "type": "sizing"
    },
    "xl": {
      "value": "{sizing.lg} * {sizing.scale}",
      "type": "sizing"
    }
  },
  "lineHeights": {
    "heading": {
      "value": "110%",
      "type": "lineHeights"
    },
    "body": {
      "value": "140%",
      "type": "lineHeights"
    }
  },
  "paragraphSpacing": {
    "h1": {
      "value": "32",
      "type": "paragraphSpacing"
    },
    "h2": {
      "value": "26",
      "type": "paragraphSpacing"
    }
  },
  "fontWeights": {
    "headingRegular": {
      "value": "Regular",
      "type": "fontWeights"
    },
    "headingBold": {
      "value": "Bold",
      "type": "fontWeights"
    },
    "bodyRegular": {
      "value": "Regular",
      "type": "fontWeights"
    },
    "bodyBold": {
      "value": "Bold",
      "type": "fontWeights"
    }
  },
  "fontSizes": {
    "h1": {
      "value": "{fontSizes.h2} * 1.25",
      "type": "fontSizes"
    },
    "h2": {
      "value": "{fontSizes.h3} * 1.25",
      "type": "fontSizes"
    },
    "h3": {
      "value": "{fontSizes.h4} * 1.25",
      "type": "fontSizes"
    },
    "h4": {
      "value": "{fontSizes.h5} * 1.25",
      "type": "fontSizes"
    },
    "h5": {
      "value": "{fontSizes.h6} * 1.25",
      "type": "fontSizes"
    },
    "h6": {
      "value": "{fontSizes.body} * 1",
      "type": "fontSizes"
    },
    "body": {
      "value": "16",
      "type": "fontSizes"
    },
    "sm": {
      "value": "{fontSizes.body} * 0.85",
      "type": "fontSizes"
    },
    "xs": {
      "value": "{fontSizes.body} * 0.65",
      "type": "fontSizes"
    }
  }
}
```

**Comment by dmackerman** [comment]
Good question. I'm in the process of evaluating Figma Tokens, and I'm super weary of using _it_ as the source of truth -- rather than having a transformer that takes a standard StyleDictionary config and goes the other way.

**Comment by djmtype** [comment]
@dmackerman I found a couple things out when using Figma Tokens with SD.  
BTW, I just started usingSD yesterday, so I'm really green. 

First `fontSizes` will need to be changed to `sizes` otherwise the default css transform which automatically appends rem as the default unit won't be applied. 

I'm hoping someone could help with a JS function whereby this wouldn't need to be fixed before compiling the JSON advance. 

Also, options: `{ outputReferences: true }` needs to be defined within the `files` object within your SD config file. 

Here's mine: 

```js
// sd.config.js

module.exports = {
	source: ['tokens/**/*.json'],
	platforms: {
		css: {
			transformGroup: 'css',
			// basePxFontSize: 16,
			buildPath: 'build/css/',
			transforms: ['attribute/cti', 'name/cti/kebab', 'color/hsl', 'size/rem'],
			files: [
				{
					destination: '_variables.css',
					format: 'css/variables',
					options: {
						outputReferences: true,
					},
				},
			],
		},
	},
};

```

I'll post anything else which might come up. 

I noticed the colors weren't transforming either. I specified HSL as the transform and I'm still getting HEX.
